pkg/sentry/platform/ptrace/subprocess_amd64.go

```diff
@@ -143,3 +145,91 @@ func (t *thread) adjustInitRegsRip() {
 func initChildProcessPPID(initregs *syscall.PtraceRegs, ppid int32) {
 	initregs.R15 = uint64(ppid)
 }
+
+// patchSignalInfo patches the signal info to account for hitting the seccomp
+// filters from vsyscall emulation, specified below. We allow for SIGSYS as a
+// synchronous trap, but patch the structure to appear like a SIGSEGV with the
+// Rip as the faulting address.
+//
+// Note that this should only be called after verifying that the signalInfo has
+// been generated by the kernel.
+func patchSignalInfo(regs *syscall.PtraceRegs, signalInfo *arch.SignalInfo) {
+	if linux.Signal(signalInfo.Signo) == linux.SIGSYS {
+		signalInfo.Signo = int32(linux.SIGSEGV)
+
+		// Unwind the kernel emulation, if any has occurred. A SIGSYS is delivered
+		// with the si_call_addr field pointing to the current RIP. This field
+		// aligns with the si_addr field for a SIGSEGV, so we don't need to touch
+		// anything there. We do need to unwind emulation however, so we set the
+		// instruction pointer to the faulting value, and "unpop" the stack.
+		regs.Rip = signalInfo.Addr()
+		regs.Rsp -= 8
+	}
+}
+
+// enableCpuidFault enable cpuid-faulting; this may fail on older kernels or hardware,
+// so we just disregard the result. Host CPUID will be enabled.
+func enableCpuidFault() {
+	syscall.RawSyscall6(syscall.SYS_ARCH_PRCTL, linux.ARCH_SET_CPUID, 0, 0, 0, 0, 0)
+}
+
+// createSeccompRules define the seccomp rules to allows only the system calls needed by
+// the stub and all its children.
+func createSeccompRules(defaultAction linux.BPFAction) []seccomp.RuleSet {
+	rules := []seccomp.RuleSet{
+		// Rules for trapping vsyscall access.
+		{
+			Rules: seccomp.SyscallRules{
+				syscall.SYS_GETTIMEOFDAY: {},
+				syscall.SYS_TIME:         {},
+				309:                      {}, // SYS_GETCPU.
```

Comment:
Is this the only difference between amd and arm? You could change it to linux.SYS_GETCPU and keep in the subprocess.go.

Reply:
No, there is another difference. The syscall.SYS_ARCH_PRCTL is not supported on arm. 
I think it would be better to move these codes to arch specific files if we need to do more customization in the future. What's your opinion?

Reply:
I'd like to keep the common part of the code in a single place, especially given the delta is small (one single syscall). This is to avoid missing one of them when making changes. If they diverge much in the future, we can consider forking the code more drastically.

For this change, you could add an arch specific call to edit the ruleset and add SYS_ARCH_PRCTL for amd64.

Reply:
Done，thanks！